annotation/tests/test_annotations.py

```diff
@@ -482,3 +581,163 @@ def test_construct_annotated_doc_db_error(
+            }
+        ],
+        2: [
+            {
+                "user_id": uuid.UUID("82aa573f-43d8-40ab-8837-282b315f7c3a"),
+                "pipeline": annotated_doc_schema.pipeline,
+                "job_id": annotated_doc_schema.job_id,
+                "file_id": annotated_doc_schema.file_id,
+                "revision": annotated_doc_schema.revision,
+                "page_id": None,
+                "date": annotated_doc_schema.date,
+                "is_validated": False,
+                "categories": annotated_doc_schema.categories,
+            },
+            {
+                "user_id": annotated_doc_schema.user,
+                "pipeline": None,
+                "job_id": 1,
+                "file_id": 1,
+                "revision": "20fe52cce6a632c6eb09fdc5b3e1594f926eea69",
+                "page_id": None,
+                "date": datetime(2024, 1, 1, 10, 10),
+                "is_validated": True,
+                "categories": {"2", "1"},
+            },
+        ],
+    }
+
+    mark_all_revisions_validated_pages(
+        page_revision_list_all, annotated_doc_schema, {1, 2}
+    )
+
+    assert page_revision_list_all == expected_pages
+
+
+def test_mark_latest_rev_validated_pages(
+    page_revision_list_latest: Dict[int, Dict[str, LatestPageRevision]],
+    annotated_doc_schema: AnnotatedDocSchema,
+):
+    expected_pages = {
+        1: {
+            uuid.UUID("6ffab2dd-3605-46d4-98a1-2d20011e132d"): {
+                "pipeline": None,
+                "job_id": 1,
+                "file_id": 1,
+                "revision": "20fe52cce6a632c6eb09fdc5b3e1594f926eea69",
+                "page_id": None,
+                "date": datetime(2024, 1, 1, 10, 10),
+                "is_validated": True,
+                "categories": {"2", "1"},
+            }
+        },
+        2: {
+            uuid.UUID("82aa573f-43d8-40ab-8837-282b315f7c3a"): {
+                "pipeline": annotated_doc_schema.pipeline,
+                "job_id": annotated_doc_schema.job_id,
+                "file_id": annotated_doc_schema.file_id,
+                "revision": annotated_doc_schema.revision,
+                "page_id": None,
+                "date": annotated_doc_schema.date,
+                "is_validated": False,
+                "categories": annotated_doc_schema.categories,
+            },
+            uuid.UUID("6ffab2dd-3605-46d4-98a1-2d20011e132d"): {
+                "pipeline": None,
+                "job_id": 1,
+                "file_id": 1,
+                "revision": "20fe52cce6a632c6eb09fdc5b3e1594f926eea69",
+                "page_id": None,
+                "date": datetime(2024, 1, 1, 10, 10),
+                "is_validated": True,
+                "categories": {"2", "1"},
+            },
+        },
+    }
+    mark_latest_revision_validated_pages(
+        page_revision_list_latest, annotated_doc_schema, {1, 2}
+    )
+    assert page_revision_list_latest == expected_pages
```

Comment:
Need to test the part where a KeyError is raised in the function  mark_latest_revision_validated_pages.

Reply:
The KeyError is not being raised in the function, it is caught and processed. The error would raise in situation when there is no page for the revision.user (page 2 in this test case).